### PR TITLE
Issue 2227 ntv flag run new templates added in specific version

### LINKED ISSUE #2227: Option to run only new template by given version
### Please describe your feature request:
Nuclei have the option to run templates added into the latest release using `-nt`,`-new-templates` cli flag which picks the templates added into the latest release, but this is limited to latest version only, and many times upon quick releases of template project it's not possible to run templates added in previous releases as new list will be updated as soon as new release has been published.

Since now nuclei template projects maintain **[.new-addition](https://github.com/projectdiscovery/nuclei-templates/blob/master/.new-additions)** file, we always have a copy of newly added templates for each version and can be added as a feature to run templates for previous/supported versions.

```
   -ntv, -new-templates-version          run new templates added in specific version
```

```
nuclei -ntv v8.8.4
nuclei -ntv 8.8.4
nuclei -ntv 8.8.4,8.8.3
```

### Describe the use case of this feature:
- Let us run newly added templates of specific versions on demand.

**Note:**

**.new-addition** file is added since v8.8.5, so validation needs to be added for min and max version.

### PR DESCRIPTION
**Pull request by LuitelSamikshya**
## Proposed changes

Closes #2227

## Checklist

- [x] Pull request is created against the [dev](https://github.com/projectdiscovery/nuclei/tree/dev) branch
- [x] All checks passed (lint, unit/integration/regression tests etc.) with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: v2/internal/runner/runner.go [hunk 1]
```diff
@@ -307,6 +311,24 @@ func (r *Runner) RunEnumeration() error {
 		}
 		r.options.Templates = append(r.options.Templates, templatesLoaded...)
 	}
+	if len(r.options.NewTemplatesWithVersion) > 0 {
+		minVersion, _ := semver.Parse("8.8.5")
```
Comment (Mzack9999) [inline v2/internal/runner/runner.go:315]: `err` should be checked and propagated if not nil

File: v2/internal/runner/runner.go [hunk 2]
```diff
@@ -307,6 +311,24 @@ func (r *Runner) RunEnumeration() error {
...
+		latestVersion, err := semver.Parse(r.templatesConfig.NucleiTemplatesLatestVersion)
+		if err != nil {
+			return errors.Wrap(err, "could not get latest version")
+		}
+		for _, version := range r.options.NewTemplatesWithVersion {
+			current, _ := semver.Parse(strings.Trim(version, "v"))
```
Comment (Mzack9999) [inline v2/internal/runner/runner.go:321]: `err` should be checked and propagated if not nil

**Review by Mzack9999 — CHANGES_REQUESTED** [review]
Please review the following points:
- [ ] `err` checks
- [ ] Check if functional tests related to this functionality can be added to `https://github.com/projectdiscovery/nuclei/blob/master/v2/cmd/functional-test/testcases.txt` (eg. `nuclei -ntv 8.8.6`)

File: v2/internal/runner/runner.go (hunk 1 above)
Comment (LuitelSamikshya) [reply v2/internal/runner/runner.go]: Done

File: v2/internal/runner/runner.go (hunk 2 above)
Comment (LuitelSamikshya) [reply v2/internal/runner/runner.go]: Done

**Review by Mzack9999 — CHANGES_REQUESTED** [review]
Version `v8.8.5` is a valid version:
```console
$ tar -tvf nuclei-templates-8.8.5.tar.gz | grep ".new-additions"
-rw-rw-r--  0 root   root     3621 Feb 18 07:38 nuclei-templates-8.8.5/.new-additions
$ go run .  -ntv 8.8.5 -u https://192.168.1.1

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   2.7.3

                projectdiscovery.io

[WRN] Use with caution. You are responsible for your actions.
[WRN] Developers assume no liability and are not responsible for any misuse or damage.
[FTL] Could not run nuclei: version should be greater than 8.8.5 and less than 9.1.0
exit status 1
```

**Review by ehsandeep — APPROVED** [review]

**Review by Mzack9999 — APPROVED** [review]

**Pull request merged**